src/submission/models.py

```diff
@@ -2141,6 +2252,53 @@ def is_correspondence_author(self):
             return True
 
 
+class CreditRecord(AbstractLastModifiedModel):
+    """Represents a CRediT record for an article"""
+
+    class Meta:
+        verbose_name = 'CRediT record'
+        verbose_name_plural = 'CRediT records'
+
+    author = models.ForeignKey(
+        'core.Account',
+        blank=True,
+        null=True,
+        on_delete=models.CASCADE,
+    )
+    frozen_author = models.ForeignKey(FrozenAuthor,
+                                      blank=True,
+                                      null=True,
+                                      on_delete=models.CASCADE)
+    preprint_author = models.ForeignKey(repository_models.PreprintAuthor,
+                                        blank=True,
+                                        null=True,
+                                        on_delete=models.CASCADE)
+    article = models.ForeignKey(Article,
+                                blank=True,
+                                null=True,
+                                on_delete=models.CASCADE)
+    role = models.CharField(max_length=100,
+                            blank=True,
+                            null=True,
+                            choices=CREDIT_ROLE_CHOICES)
+
+    def __str__(self):
+        return self.get_role_display()
+
+    @property
+    def uri(self):
+        return f"https://credit.niso.org/contributor-roles/{self.role}/"
+
+    @staticmethod
+    def all_roles(self):
+        return CREDIT_ROLE_CHOICES
+
+    def save(self, *args, **kwargs):
+        exclusive_fields = ['author', 'frozen_author', 'preprint_author']
+        model_utils.validate_exclusive_fields(self, fields=exclusive_fields)
+        super().save(*args, **kwargs)
```

Comment:
When I suggested checking for mutual exclusion, I meant as a database constraint, as models and querysets have other ways to write to db. e.g:

```py
class Meta:
        constraints = [
            models.CheckConstraint(
                check=(
                    models.Q(author__isnull=True, frozen_author__isnull=True, preprint_author__isnull=False) |
                    models.Q(author__isnull=True, frozen_author__isnull=False, preprint_author__isnull=True) |
                    models.Q(author__isnull=False, frozen_author__isnull=True, preprint_author__isnull=True)
                ),
                name='single_author_relation_constraint'
            )
        ]
```

For extra fancy points, you could generalize this approach for your list of `exclusive_fields`